framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -205,6 +225,32 @@ public void updateForMethodCall(
             MethodInvocationNode n, AnnotatedTypeFactory atypeFactory, V val) {
         ExecutableElement method = n.getTarget().getMethod();
 
+        // List of expressions that this method side-effects (specified as arguments/elements of
+        // @SideEffectsOnly). If the list is empty, then there is no @SideEffectsOnly annotation.
+        List<JavaExpression> sideEffectsOnlyExpressions = new ArrayList<>();
+        if (isSideEffectsOnly(atypeFactory, method)) {
+            SourceChecker checker = analysis.checker;
+            JavaExpressionParseUtil.JavaExpressionContext methodUseContext =
+                    JavaExpressionParseUtil.JavaExpressionContext.buildContextForMethodUse(
+                            n, checker);
+
+            AnnotationMirror sefOnlyAnnotation =
+                    atypeFactory.getDeclAnnotation(method, SideEffectsOnly.class);
+            List<String> sideEffectsOnlyExpressionStrings =
+                    AnnotationUtils.getElementValueArray(
+                            sefOnlyAnnotation, "value", String.class, true);
+
+            for (String st : sideEffectsOnlyExpressionStrings) {
+                try {
+                    JavaExpression exprJe = JavaExpressionParseUtil.parse(st, methodUseContext);
+                    sideEffectsOnlyExpressions.add(exprJe);
+                } catch (JavaExpressionParseUtil.JavaExpressionParseException ex) {
+                    System.out.println(ex.getDiagMessage());
```

Comment:
This is not the right way to report errors.  Please use the standard Checker Framework mechanisms.